src/app/content/highlights/components/Highlights.tsx

```diff
@@ -39,11 +49,25 @@ const Highlights = ({ className }: { className: string }) => {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [orderedHighlights]);
 
+  React.useEffect(() => {
+    if (
```

Comment:
The conditions here are the same as the conditions determining what's returned. It seems like the returned items should each be their own component with a name that tells readers what's going on. Each of those components can then set its own announceMsg rather than needing a state variable for it.